src/main/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationController.java

```diff
@@ -87,6 +93,32 @@ public PublicationController(PublicationService publicationService) {
         return ResponseEntity.status(HttpStatus.CREATED).body(createdItem);
     }
 
+    @ApiResponses({
+        @ApiResponse(code = 200,
+            message = "List of Artefacts matching the given courtId and verification parameters and date requirements"),
+        @ApiResponse(code = 404,
+            message = "No artefact found matching given parameters and date requirements"),
+    })
+    @ApiOperation("Get a series of publications matching a given input (e.g. courtid)")
+    @GetMapping("/search/{searchValue}")
+    public ResponseEntity<List<Artefact>> getAllRelevantArtefactsByCourtId(@PathVariable String searchValue,
+                                                                           @RequestHeader Boolean verification) {
+        return ResponseEntity.ok(publicationService.findAllByCourtId(searchValue, verification));
+    }
+
+    @ApiResponses({
+        @ApiResponse(code = 200,
+            message = "Blob data from the given request in text format.",
+            response = String.class),
+        @ApiResponse(code = 404,
+            message = "No artefact found matching given parameters and date requirements"),
+    })
+    @ApiOperation("Get the info from within a blob given source artefact id and provenance")
+    @GetMapping("/{artefactId}")
+    public ResponseEntity<String> getBlobData(@PathVariable UUID artefactId) {
```

Comment:
Just been trying this out. Looks good! I think we'll want the verified flag on the getBlobData endpoint too in the header, just to make sure that the user requesting the data from the blob is verified